File: src/2236-Root-Equals-Sum-of-Children/root_equals_sum_children.cpp
#include <iostream>

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    bool checkTree(TreeNode* root) {
        if(root->val == (root->left->val + root->right->val)) return true;
        else return false;
    }
};

int main() {
    TreeNode left_child = TreeNode(4);
    TreeNode right_child = TreeNode(6);
    TreeNode root = TreeNode(10, &left_child, &right_child);
    Solution solution;
    std::cout << solution.checkTree(&root);

    return 0;
}

File: src/2236-Root-Equals-Sum-of-Children/root_equals_sum_children.js
// Definition for a binary tree node.
function TreeNode(val, left, right) {
  this.val = (val===undefined ? 0 : val)
  this.left = (left===undefined ? null : left)
  this.right = (right===undefined ? null : right)
 }

/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var checkTree = function(root) {
  return root.val == root.left.val + root.right.val
}

left_child = new TreeNode(4)
right_child = new TreeNode(6)
root = new TreeNode(10, left_child, right_child)
console.log(checkTree(root))